config.go

```diff
@@ -148,10 +149,11 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin := range pluginPaths {
+	for plugin, path := range pluginPaths {
 		plugin := plugin
+		path := path
 		c.PostProcessors[plugin] = func() (packer.PostProcessor, error) {
-			return c.pluginClient(pluginPaths[plugin]).PostProcessor()
+			return c.pluginClient(path).PostProcessor()
```

Comment:
Hey @TJM, thanks for opening; to me this is doing exactly the same thing as before; did this fix your issue ?

`plugin := plugin` was making the plugin variable local and therefore the range loop would not overwrite it. `pluginPaths` never changes.

Here's an edit of your previous snippet that shows it: https://play.golang.org/p/KM88f-Z69xe

Reply:
Yes, my code fixes the problem with the path. The example code doesn't cover where pluginPaths is being overwritten because it processes 3 different types of plugins: https://github.com/hashicorp/packer/blob/master/config.go#L130-L164

Presumably, because provisioner is the last type "discovered" it would still have worked, but builder and post-processor do not.

Tommy

Reply:
Here is an example https://play.golang.org/p/WJ710aIBWIN

Reply:
Ah right ! thanks for the explanation !